arch/rv32i/src/pmp.rs

```diff
@@ -352,8 +354,18 @@ impl kernel::mpu::MPU for PMPConfig {
         csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::r0::SET);
         csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::w0::SET);
         csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::x0::SET);
-        csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::a0::TOR)
+        csr::CSR.pmpcfg0.modify(csr::pmpconfig::pmpcfg::a0::TOR);
+        // Set the dirty flag
+        self.is_dirty.set(true);
     }
+}
+
+impl kernel::mpu::MPU for PMPConfig {
+    type MpuConfig = PMPConfig;
+
+    fn enable_mpu(&self) {}
+
+    fn disable_mpu(&self) {}
```

Comment:
This came up in: https://github.com/tock/tock/pull/1871 and isn't the right approach as people might still want to disable PMP at boot.

Reply:
It also means we now don't match the trait documentation.

Reply:
I think there is some room for improvement in the trait. The doc says that this will be called before returning to the kernel, so I think it is reasonable to implement it as NOP for risc-v similar to `enable_mpu`, but the name is too general so it amy be misleading.

Reply:
There is a pending PR to update the trait which will then stop calling disable on every transition.